packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -159,6 +182,22 @@ function listenClickEvents(callback: (clickEvent: MouseEvent & { target: Element
   )
 }
 
+const enum PotentialClickActionStatus {
+  // Initial state, the potential click action is still ongoing.
+  PENDING,
+  // The potential click action is no more ongoing but still needs to be validated or discarded.
+  STOPPED,
+  // Final state, the potential click action has been stopped and validated or discarded.
+  FINALIZED,
+}
+
+type PotentialClickActionState =
+  | { status: PotentialClickActionStatus.PENDING }
+  | { status: PotentialClickActionStatus.STOPPED; endTime?: TimeStamp }
+  | { status: PotentialClickActionStatus.FINALIZED }
+
+export type PotentialClickAction = ReturnType<typeof newPotentialClickAction>
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏it seems to me that some namings are equivalent or quite close:

- `finalize` VS `flush`
- `potential` VS `pending`

Proposal:
- use `finalize` everywhere instead of `flush`
- remove `potential` prefix and rename existing `ClickAction` to `ValidatedClickAction`

wdyt?

Reply:
Ok to rename `flush` to `finalize` in the clickchain

For removing the `potential` prefix, I'm not sure... I think having the `potential` prefix is a bit easier to understand that it won't necessarily generate an Action. Also, potential/validated concept is a `trackClickAction` implementation detail... it would be nice if it's not leaking to other parts of the SDK.